zebrad/src/commands/start.rs

```diff
@@ -315,9 +320,53 @@ impl StartCmd {
         // TODO: replace with `MAX_CLOSE_TO_TIP_BLOCKS` after fixing slow syncing near tip (#3375)
         const MIN_SYNC_WARNING_BLOCKS: i32 = 60;
 
+        // The number of fractional digits in sync percentages.
+        const SYNC_PERCENT_FRAC_DIGITS: usize = 3;
+
+        // The minimum number of extra blocks after the highest checkpoint, based on:
+        // - the non-finalized state limit, and
+        // - how long it takes to build Zebra and re-sync non-finalized blocks (~12 minutes).
```

Comment:
Question: I don't understand this part. This is related to the `+ 10` part of the equation, right? Is this some sort of margin of error? I also don't understand how the time to build Zebra relates to it.

Reply:
I think I now understand what it is :tada: 

But I still don't understand why we need it :thinking: What can go wrong if we test it without the margin and right after a checkpoint has been committed?

Reply:
If the full sync hangs or panics when we commit a block from the non-finalised state to the finalized state, having this extra margin helps catch that bug. (Either in a local cached tip test, or if we add a cached tip test to CI.)

Otherwise the full sync might never run the finalisation code. Particularly if there is some clock skew on the miner or the local node.

I guess I should say that in the comments.

Reply:
Fixed in 78f610d515a22811180af6f7e4845197f0eed2da

Reply:
Awesome, thanks!